python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -432,73 +433,94 @@ def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
+                matcher_struct = pyPPMb.match(
+                    source_array=src_array,
+                    n_check=self.config.numPointsForShapeAttempt + try_idx,
+                    n_match=self.config.numPointsForShape,
+                    n_agree=run_n_consent,
+                    max_n_patterns=self.config.numBrightStars,
+                    max_shift=maxShiftArcseconds,
+                    max_rotation=self.config.maxRotationDeg,
+                    max_dist=maxMatchDistArcSec * 2. ** soften_idx,
+                    min_matches=minMatchedPairs,
+                    pattern_skip_array=np.array(
+                        match_tolerance.failedPatternList)
+                )
+
+                if try_idx == 0 and soften_idx == 0 and \
+                   len(matcher_struct.match_ids) == 0 and \
+                   match_tolerance.lastMatchedPattern is not None:
+                    # If we found a pattern on a previous match-fit iteration and
+                    # can't find an optimistic match on our first try with the
+                    # tolerances as found in the previous match-fit,
+                    # the match we found in the last iteration was likely bad. We
+                    # append the bad match's index to the a list of
+                    # patterns/matches to skip on subsequent iterations.
+                    match_tolerance.failedPatternList.append(
+                        match_tolerance.lastMatchedPattern)
+                    match_tolerance.lastMatchedPattern = None
+                    maxShiftArcseconds = \
+                        self.config.maxOffsetPix * wcs.pixelScale().asArcseconds()
+                    continue
+                elif len(matcher_struct.match_ids) > 0:
+                    # Match found, save a bit a state regarding this pattern
+                    # in the match tolerance class object and exit.
+                    match_tolerance.maxShift = \
+                        matcher_struct.shift * afwgeom.arcseconds
+                    match_tolerance.lastMatchedPattern = \
+                        matcher_struct.pattern_idx
+                    match_found = True
+                    break
+            if match_found:
                 break
 
+        # If we didn't find a match, exit early.
+        if not match_found:
+            return pipeBase.Struct(
+                matches=[],
+                match_tolerance=match_tolerance,
+            )
+
         # The matcher returns all the nearest neighbors that agree between
         # the reference and source catalog. For the current astrometric solver
         # we need to remove as many false positives as possible before sending
         # the matches off to the solver.
-        distances_arcsec = np.degrees(matcher_struct.distances_rad) * 3600
+        distances_arcsec = np.degrees(matcher_struct.distances_rad) * 3600 
+        clip_max_dist = np.max(
+            (sigmaclip(distances_arcsec, low=100, high=2)[-1],
```

Comment:
What's the justification for these values of `low` and `high`? They just seem like magic numbers here.

Reply:
The are the number of sigma for the clip. Low is 100 because the lower bound is always a value of 0 and high is a 2 sigma cut. I will add a comment explaining this. I could also make the high value a configurable if that is useful.